libraries/AP_HAL_ChibiOS/hwdef/CUAV-7-Nano/hwdef.dat

```diff
@@ -0,0 +1,294 @@
+# hw definition file for processing by chibios_hwdef.py
+# for CUAV-7-Nano board
+
+# default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 2
+
+# board ID for firmware load
+APJ_BOARD_ID 7000
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# to be compatible with the px4 bootloader we need
+# to use a different RAM_MAP
+env USE_ALT_RAM_MAP 1
+
+# flash size
+FLASH_SIZE_KB 2048
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 UART7 UART5 USART1 UART8 USART3 OTG2
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PA9  VBUS INPUT OPENDRAIN
+
+# pins for SWD debugging
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# telem1
+PE8  UART7_TX UART7
+PF6  UART7_RX UART7
+PF8  UART7_RTS UART7
+PE10 UART7_CTS UART7
+
+# telem2
+PC8  UART5_RTS UART5
+PC9  UART5_CTS UART5
+PC12 UART5_TX UART5
+PD2  UART5_RX UART5
+
+# GPS1
+PB6  USART1_TX USART1
+PB7  USART1_RX USART1
+
+# GPS2
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# debug uart
+PD8  USART3_TX USART3
+PD9  USART3_RX USART3
+
+# uart6, RX only, RC input, if no IOMCU
+PC7  USART6_RX USART6
+
+# ethernet (not implemented yet)
```

Comment:
I just received a Nano-7 (and I love it!) 
the ethernet does work great with master, we can remove the "not implemented yet" comment